lib/src/hydrated.dart

```diff
@@ -58,48 +46,59 @@ typedef HydrateCallback<T> = T Function(String);
 ///   );
 /// ```
 class HydratedSubject<T> extends Subject<T> implements ValueStream<T> {
-  static final _areTypesEqual = TypeUtils.areTypesEqual;
   final BehaviorSubject<T> _subject;
-  final String _key;
+
+  @deprecated
+  // ignore: unused_field
   final HydrateCallback<T>? _hydrate;
+
+  @deprecated
+  // ignore: unused_field
   final PersistCallback<T>? _persist;
+
   final VoidCallback? _onHydrate;
   final T? _seedValue;
 
+  final GenericValuePersistence<T> _persistence;
+
   HydratedSubject._(
-    this._key,
     this._seedValue,
     this._hydrate,
     this._persist,
     this._onHydrate,
     this._subject,
+    this._persistence,
   ) : super(_subject, _subject.stream) {
     _hydrateSubject();
   }
 
-  factory HydratedSubject(
-    String key, {
+  factory HydratedSubject({
+    @Deprecated('`key` is deprecated. '
+        'Please use `persistence` for managing storage location.')
+        String? key,
```

Comment:
A breaking change:
Turned the `key` required positional arg into an optional named arg.

This was done to avoid problems for the users of the updated API. 
Without this change, it would look like this -- a duplicated `key` argument:
```dart
  final _countSubject = HydratedSubject<SerializedClass>(
    key: "serialized-count",
    persistence: SharedPreferencesPersistence(
      key: "serialized-count",
      hydrate: (value) => SerializedClass.fromJSON(value),
      persist: (value) => value.toJSON,
    ),
    seedValue: SerializedClass(0),
  );
```